runChecks.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
+
+"""
+Replacement for test-math-dependencies target in Makefile.
+
+"""
+
+from __future__ import print_function
+import os
+import os.path
+import sys
+import re
+import glob
+
+winsfx = ".exe"
+testsfx = "_test.cpp"
+def files_in_folder(folder):
+    files = []
+    for f in glob.glob(folder):
+        if os.path.isdir(f):
+            files += files_in_folder(f+os.sep+"**")
+        else:
+            files.append(f)
+    return files
+def grep_patterns(type, folder, patterns_and_messages, exclude_filters = []):
+    """Checks the files in the provided folder for matches
+    with any of the patterns. It returns an array of
+    messages and the provided type with
+    the line number. This check ignores comments.
+    """
+    errors = []
+    folder.replace("/", os.sep)
+    exclude_files = []
+    for excl in exclude_filters:
+        exclude_files += files_in_folder(excl)
+    files = files_in_folder(folder+os.sep+"**")
+    files = [x for x in files if x not in exclude_files]
+    for f in files:
+        if os.path.isfile(f):
+            line_num = 0
+            for line in open(f, "r"):
+                line_num += 1
+                
+                for p in patterns_and_messages:
+                    # exclude line starting with "/*" or " * " and
+                    # if the matched patterns are behind "//"
+                    if not re.search("^ \* |^/\*", line) and not re.search(".*//.*"+p["pattern"], line) and re.search(p["pattern"], line):
```

Comment:
I have this monster in ArviZ to remove comments, but not sure if it would even work here

```
pattern_remove_comments = re.compile(
        r'//.*?$|/\*.*?\*/|\'(?:\\.|[^\\\'])*\'|"(?:\\.|[^\\"])*"', re.DOTALL | re.MULTILINE
    )
...
stan_code = re.sub(pattern_remove_comments, "", stan_code)
```

https://github.com/arviz-devs/arviz/blob/cdbe656a0c759c3f0d83e7285b8b7e8ac5992704/arviz/data/io_pystan.py#L511

Reply:
But your current code does not handle multiline comments.

So you could find the locations for the comments and then eliminate stuff based on those limits (start, end)

Reply:
These are for `.stan` files, but I think the same comment system is used with C++?

Reply:
Yeah, the original script we are replacing here only worked with Doxygen blocked comments in the form of:
```
/**
 * text text
 */
```
I actually think that there are no block comments in the form
```
/*
 text text
 more text
*/
in the Stan Math codebase. But yes, we should also cover that, good call. Will redo this, but avoiding removing the comments to preserved the count. Thank you for the example.